jaxley/modules/base.py

```diff
@@ -118,7 +118,7 @@ def __init__(self):
         self.total_nbranches: int = 0
         self.nbranches_per_cell: List[int] = None
 
-        self.groups = {}
+        self.group_names: List[str] = []
```

Comment:
I see the reasons for keeping track of group names outside of `nodes`, but I slightly prefer the old `groups` style dictionary.